### PR TITLE
Unpack grouped tests

### PR DESCRIPTION
**Pull request by ubaumann**
Did some tests and tried something. It is not perfect because now the context has to be created two times but to be able to get all grouped hosts, I need the Nornir object.

Here I have a playground: https://github.com/ubaumann/nuts-experiments

### CODE REVIEW

File: docs/source/tutorial/firststeps.rst [hunk 1]
```diff
@@ -173,7 +175,7 @@ The sample test bundle above requires a network inventory and a running network
 
 How it works: Each test module implements a context class to provide module-specific functionality to its tests. This context class is a  ``NutsContext`` or a subclass of it. This guarantees a consistent interface across all tests for test setup and execution. 
 
-The predefined test classes which depend on a network all use `nornir <https://nornir.readthedocs.io/en/latest/>`__ in order to communicate with the network devices. Those test classes all derive all from a more specific ``NornirNutsContext``, which provides a nornir instance and nornir-specific helpers. 
+The predefined test classes which depend on a network all use `nornir <https://nornir.readthedocs.io/en/latest/>`__ in order to communicate with the network devices. Those test classes all derive from a more specific ``NornirNutsContext``, which provides a nornir instance and nornir-specific helpers.
```
Comment (marcom4rtinez) [inline docs/source/tutorial/firststeps.rst:178]: is that still true with the merge of the SA in 3.1?

File: nuts/helpers/filters.py [hunk 2]
```diff
@@ -2,12 +2,51 @@
...
+from nornir.core.filter import F, OR
 
-from nornir.core.filter import F
+from nuts.helpers.errors import NutsSetupError
 
 
 def filter_hosts(test_data: Optional[List[Dict[str, Any]]]) -> F:
     assert test_data is not None
-    hosts: Set[str] = {entry["host"] for entry in test_data}
+    hosts: List[str] = list({entry["host"] for entry in test_data})
```
Comment (marcom4rtinez) [inline nuts/helpers/filters.py:13]: Is it desired to have the same host multiple times in the same list. How is it prevented now?

Comment (ubaumann) [reply nuts/helpers/filters.py:13]: It is still a set comprehension but in the end, it is converted to a list for consistency.

File: docs/source/tutorial/firststeps.rst (hunk 1 above)
Comment (ubaumann) [reply docs/source/tutorial/firststeps.rst:178]: Good point. I need to check the tests from the SA

**Pull request merged**